.travis.yml

```diff
@@ -32,38 +32,70 @@ jobs:
       #if: tag =~ /^release.*$/
       language: shell
       services: docker
+      install:
+        # set up awscli packages
+        - python3 -m pip install --user awscli
+      before_script:
+        # set up shared storage for build artifacts
+        - mkdir -p $TRAVIS_BUILD_SHARED_STORAGE
       script:
         - docker run --rm -v `pwd`:/var/task amazon/aws-sam-cli-build-image-python3.8 bash -c './build-release-package.sh'
-        - cp -p dynatrace-aws-log-forwarder.zip ~/build-stages-shared-storage/$TRAVIS_BUILD_NUMBER
+      after_success:
+        - cp -p dynatrace-aws-log-forwarder.zip $TRAVIS_BUILD_SHARED_STORAGE
+        - aws s3 sync $TRAVIS_BUILD_SHARED_STORAGE $AWS_S3_BUILD_SHARED_STORAGE
     - stage: e2e_tests
       name: Execute e2e tests
       if: branch = master OR type = pull_request
       language: shell
       services: docker
-      before_script: 
-        - mkdir -p ./e2e_tests_tmp
-        - cp -p ~/build-stages-shared-storage/$TRAVIS_BUILD_NUMBER/dynatrace-aws-log-forwarder.zip ./e2e_tests_tmp
-        - cd ./e2e_tests_tmp
+      install:
+        # set up awscli packages
+        - python3 -m pip install --user awscli
+        # set up e2e packages
+        - python3 -m pip install --user -r ./tests/e2e/requirements.txt
+      before_script:
+        # set up shared storage for build artifacts
+        - mkdir -p $TRAVIS_BUILD_SHARED_STORAGE
+        - aws s3 sync $AWS_S3_BUILD_SHARED_STORAGE $TRAVIS_BUILD_SHARED_STORAGE
+        - cp -p $TRAVIS_BUILD_SHARED_STORAGE/dynatrace-aws-log-forwarder.zip ./tests/e2e
+        - cd ./tests/e2e
         - unzip dynatrace-aws-log-forwarder.zip
       script:
         # deploy log forwarder resources reqiured for e2e to AWS
-        - ./dynatrace-aws-logs.sh deploy --target-url $DYNATRACE_ENV_URL --target-api-token $DYNATRACE_API_KEY --use-existing-active-gate true --require-valid-certificate $VERIFY_SSL --stack-name aws-log-forwarder-e2e-build-$TRAVIS_BUILD_NUMBER
+        - ./dynatrace-aws-logs.sh deploy --target-url $DYNATRACE_ENV_URL --target-api-token $DYNATRACE_API_KEY --use-existing-active-gate true --require-valid-certificate $VERIFY_SSL --stack-name $AWS_CLOUDFORMATION_STACK

```

Comment:
I personally like exporting myself those repeating dynatrace-aws-logs variables 
```
export STACK_NAME=BeluPerfLogsCLientSecond
# more exports
./dynatrace-aws-logs.sh deploy --use-existing-active-gate true

export LOG_GROUPS_LIST="/aws/lambda/BeluAWSLogsPerformanceTest_inputLogsProducingLambda"
./dynatrace-aws-logs.sh subscribe
```

I thought maybe you could set them once here or even call the global like that already and it could work. 

We get more brevity, maybe a little bit more readability because of it, but for the price of it maybe not being explicit from this file here where the variable is used. Just a suggestion, but this is totally optional.

Reply:
Command line arguments is the main deployment way from the docs and for me this would be a reason to stick to that.
If for some reason deployment with exports does not work then its not as big as a problem as if main deployment way with cmd args stops working

Reply:
Ok, forgot there's anything in between that could go wrong.